realm/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -304,4 +316,33 @@ public void testHandlerNotRemovedToSoon() {
         instance1.close();
         assertNull(instance1.getHandler());
     }
+
+    public void testImmediateNotificationsOnSameThread() {
+        final AtomicBoolean success = new AtomicBoolean(false);
+        realm = Realm.getInstance(getContext());
+        realm.addChangeListener(new RealmChangeListener() {
+            @Override
+            public void onChange() {
+                success.set(true);
+            }
+        });
+        realm.beginTransaction();
+        realm.createObject(AllTypes.class);
+        realm.commitTransaction();
+        assertTrue(success.get());
+    }
+
+    public void testEmptyCommitTriggerOnChangeHandler() {
```

Comment:
Confusing test name. We don't use the word handler anywhere else, unless to specify the Android's Handler class